sfge/src/test/java/com/salesforce/graph/ops/MethodUtilTest.java

```diff
@@ -0,0 +1,282 @@
+                TestUtil.Config.Builder.get(g, new String[] {SOURCE_FILE_1, SOURCE_FILE_2}).build();
+        TestUtil.buildGraph(config);
+        // Create a rule target encompassing only the method in the first file whose name is used
+        // elsewhere.
+        List<RuleRunnerTarget> targets = new ArrayList<>();
+        targets.add(
+                new RuleRunnerTarget(
+                        "TestCode0",
+                        Collections.singletonList(METHOD_WITH_EXTERNAL_NAME_DUPLICATION)));
+
+        List<MethodVertex> methodVertices = MethodUtil.getTargetedMethods(g, targets);
+
+        MatcherAssert.assertThat(methodVertices, hasSize(equalTo(2)));
+        MethodVertex firstVertex = methodVertices.get(0);
+        assertEquals(METHOD_WITH_EXTERNAL_NAME_DUPLICATION, firstVertex.getName());
+        assertEquals(18, firstVertex.getBeginLine());
+
+        MethodVertex secondVertex = methodVertices.get(1);
+        assertEquals(METHOD_WITH_EXTERNAL_NAME_DUPLICATION, secondVertex.getName());
+        assertEquals(22, secondVertex.getBeginLine());
+
+        String messages = CliMessager.getInstance().getAllMessages();
+        MatcherAssert.assertThat(
+                messages,
+                containsString(EventKey.WARNING_MULTIPLE_METHOD_TARGET_MATCHES.getMessageKey()));
+    }
+
+    @Test
+    public void getTargetMethods_targetMethodDoesNotExist() {
+        TestUtil.Config config =
+                TestUtil.Config.Builder.get(g, new String[] {SOURCE_FILE_1, SOURCE_FILE_2}).build();
+        TestUtil.buildGraph(config);
+        // Create a rule target encompassing a method that doesn't actually exist with that name.
+        List<RuleRunnerTarget> targets = new ArrayList<>();
+        targets.add(
+                new RuleRunnerTarget(
+                        "TestCode0", Collections.singletonList(METHOD_THAT_DOES_NOT_EXIST)));
+
+        List<MethodVertex> methodVertices = MethodUtil.getTargetedMethods(g, targets);
+
+        MatcherAssert.assertThat(methodVertices, hasSize(equalTo(0)));
+
+        String messages = CliMessager.getInstance().getAllMessages();
+        MatcherAssert.assertThat(
+                messages,
+                containsString(EventKey.WARNING_NO_METHOD_TARGET_MATCHES.getMessageKey()));
+    }
+
+	@Test
+	public void getTargetMethods_targetMethodInInnerClass() {
+		TestUtil.Config config =
+			TestUtil.Config.Builder.get(g, new String[]{SOURCE_FILE_3}).build();
+		TestUtil.buildGraph(config);
+		// Create a rule target encompassing the method that exists only in the inner class.
+		List<RuleRunnerTarget> targets = new ArrayList<>();
+		targets.add(new RuleRunnerTarget("TestCode0", Collections.singletonList(METHOD_WITHOUT_OVERLOADS_3)));
+
+		List<MethodVertex> methodVertices = MethodUtil.getTargetedMethods(g, targets);
+
+		MatcherAssert.assertThat(methodVertices, hasSize(equalTo(1)));
+	}
+
+	@Test
+	public void getTargetMethods_targetMethodInInnerAndOuterClass() {
```

Comment:
Test case: A method that exists in both inner and outer class. QUESTION: The syntax here would be `path/to/file.cls#methodName`, and it would return both the inner and outer class methods. Are we okay with that? I'm okay with that for now, and we can change it if people complain.